src/components/app/App.jsx

```diff
@@ -1,5 +1,23 @@
-import React from 'react';
+/* eslint-disable max-len */
+import React, { useReducer } from 'react';
+import { reducer, initialState } from '../reducers/useReducer';
 
 export default function App() {
-  return <h1>Hello World</h1>;
+  const [state, dispatch] = useReducer(reducer, initialState);
+
+  return (
+    <>
+      <button onClick={() => dispatch({ type: 'UNDO' })}>undo</button>
+      <button onClick={() => dispatch({ type: 'REDO' })}>redo</button>
```

Comment:
Needs to not let the user advance passed the defend variables in state.befor or state.after there are two ways of doing this

disable the buttons by
`<button disabled={!before.length}  onClick={() => dispatch({ type: 'UNDO' })}>undo</button>`
this will disable the button when there is nothing in the array
Or in the reducer
`case 'UNDO':
      return {
        ...state,
        undo: state.undo.slice(0, -1),
        redo: [state.current, ...state.redo],
        current: state.undo[state.undo.length - 1] || [#FF00FF]
      };`
If state is undefined it will set it to a default color